DSharpPlus.Core/Rest/RestClientOptions.cs

```diff
@@ -0,0 +1,22 @@
+using Microsoft.Extensions.Options;
+
+namespace DSharpPlus.Core.Rest
+{
+    /// <summary>
+    /// Stores options for the D#+ Rest Client.
+    /// </summary>
+    public class RestClientOptions : IOptions<RestClientOptions>
```

Comment:
Again with implementing IOptions. Is this to avoid DI or something?

Reply:
as above - yes, it's to give people the option to avoid DI